lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -450,6 +466,55 @@ function validatePolicyResource(bucketName, policy) {
     });
 }
 
+function checkIp(value) {
+    const isValid = /^(\d{1,3}\.){3}\d{1,3}(\/\d{1,2})?$/.test(value);
+    if (isValid) {
+        return null;
+    }
+    return 'Invalid IP address in Conditions';
+}
+
+// This function checks all bucket policy conditions if the values provided
+// are valid for the condition type. If not it returns a relevant Malformed policy error string
+function validatePolicyConditions(policy) {
+    const validConditions = [
+        { conditionKey: 'aws:SourceIp', conditionValueTypeChecker: checkIp },
+        { conditionKey: 's3:object-lock-remaining-retention-days' },
+    ];
+    // keys where value type does not seem to be checked by AWS:
+    // - s3:object-lock-remaining-retention-days
+
+    // there can be multiple statements in the policy, each with a Condition enclosure
+    for (let i = 0; i < policy.Statement.length; i++) {
+        const s = policy.Statement[i];
+        if (s.Condition) {
+            const conditionOperators = Object.keys(s.Condition);
+            // there can be multiple condition operations in the Condition enclosure
+            for (let j = 0; j < conditionOperators.length; j++) {
+                const conditionOperator = conditionOperators[j];
+                const conditionKey = Object.keys(s.Condition[conditionOperator])[0];
+                const conditionValue = s.Condition[conditionOperator][conditionKey];
+                const validCondition = validConditions.find(validCondition =>
+                    validCondition.conditionKey === conditionKey
+                );
+                // this is the seen behaviour on AWS... don't ask me why
```

Comment:
The comment could be more instructive explaining what is the AWS behavior that we reproduce here, if it is not obvious, e.g.

```suggestion
                // AWS returns an error if the condition key doesn't start with `aws:`, so we reproduce this behavior here
```